---
title: "Mouse-human alignment of transcriptomic signatures"
output: html_document
teaching: 30
exercises: 10
questions:
- "How well do transcriptomic changes we observe in mouse models carrying AD-related mutations align with human AD data?"
- "How do we perfrom a cross-species comparison"
objectives:
- "Understand the human AD modules."
- "Approach to align mouse data to human data"
- "Perform correlation analysis."
- "Visualize the results"
keypoints:
- "AMP-AD gene modules represent major transcriptomic heterogeneity in AD."
- "Correlation of logFC is a practical approach for human-mouse alignment of AD-associated transcriptomic signatures."
---

Author: Ravi Pandey, Jackson Laboratory


```{r setup, include=FALSE}
source("../bin/chunk-options.R")
knitr_fig_path("04-")
```


```{r libs, warning=FALSE, include=FALSE}
suppressPackageStartupMessages(library("DESeq2"))
suppressPackageStartupMessages(library("ggplot2"))
suppressPackageStartupMessages(library("AnnotationDbi"))
suppressPackageStartupMessages(library("org.Mm.eg.db"))
suppressPackageStartupMessages(library("GO.db"))
suppressPackageStartupMessages(library("EnhancedVolcano"))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(clusterProfiler))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(synapser))
synLogin(silent=TRUE)
#setwd("/Users/pandera/Library/CloudStorage/Box-Box/Ravi Pandey Workspace/Miscellaneous/AD_Workshop2023/")
```

## Aligning Human and Mouse Phenotype

Alzheimer's Disease (AD) is complex disease, we do not expect these mouse models to have complete late onset AD (LOAD) pathology. To study AD, we utilize MRI and PET imaging to match  back to the human imaging study [ADNI](https://adni.loni.usc.edu/), neuropathology information, and lots of genomics, proteomics, and metabolomics. These omics studies allow us to do real direct homology comparison between human and mouse since a significant portion of genes are  shared between these two species.


![Aligning Human and Mouse Phenotype](../fig/Picture3.png)


## Overview of Human transcriptomic data
There are three independent human brain transcriptome studies. They are the Religious Orders Study and the Memory and Aging Project [(ROSMAP)](https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3219045), Mount Sinai Brain Bank [(MSBB)](https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn3159438), and [Mayo RNAseq Study](https://adknowledgeportal.synapse.org/Explore/Studies/DetailsPage/StudyDetails?Study=syn5550404). Collected human postmortem brain RNA-seq data from seven distinct regions: dorsolateral prefrontal cortex (DLPFC), temporal cortex (TCX), inferior frontal gyrus (IFG), superior temporal gyrus (STG), frontal pole (FP), parahippocampal gyrus (PHG), and cerebellum (CBE), 

These postmortem samples are generally balanced for AD, MCI, and non-effected controls. This  provides a broad assessment how AD has affected multiple brain regions in 3 different populations around the US.


![Transcriptomic of Alzheimer's Disease](../fig/Picture2.png)


## Overview of Human Consensus RNA-Seq Coexpression Modules

The Accelerating Medicines Partnership-Alzheimerâ€™s Disease [(AMP-AD Consortium)](https://adknowledgeportal.synapse.org/Explore/Programs/DetailsPage?Program=AMP-AD)  has generated RNA-seq profiles from more than 1,200 human brains and is applying systems biology approaches toward the goal of elucidating AD mechanisms and potential therapeutic targets.

[Wan, et al.](https://doi.org/10.1016/j.celrep.2020.107908) performed meta analysis including all available AMP-AD RNA-seq datasets and systematically defined correspondences between gene expression changes associated with AD in human brains. Briefly, [Wan, et al.](https://doi.org/10.1016/j.celrep.2020.107908) performed library normalization and covariate adjustments for each human study separately using fixed/mixed effects modeling to account for batch effects. Among the 2978 AMP-AD modules identified across all tissues, 660 modules were selected which showed an enrichment for at least one AD-specific differential expressed gene set from the meta-analysis in cases compared to controls. 

Next, they performed multi method co-expression network analysis followed by differential analysis and found 30 co-expression modules related to LOAD pathology from a human cohort study. Among the 30 aggregate co-expression modules, five consensus clusters have been described by [Wan, et al.](https://doi.org/10.1016/j.celrep.2020.107908) These consensus clusters consist of a subset of modules which are associated with similar AD related changes across the multiple studies and brain regions. Further, they looked for enrichment of cell type signatures in these modules using expression-weighted cell type enrichment analysis [Skene and Grant, 2016](https://doi.org/10.3389/fnins.2016.00016) as well as  applied functional annotations to these modules.

The first module block is enriched in astrocytes, the next block is enriched in endothelial and microglial genes suggesting strong inflammation component, the third block in strongly enriched in neuron suggesting neurodegeneration, fourth is enriched in oligodendrocytes and glial genes suggesting myelination and finally mixed modules that have things to do like stress response and response to unfolded proteins. Stress response is not cell specific, so they may be throughout many cells in brain.


![AMP-AD GENE Modules](../fig/Picture1.png)

Here we are showing a matrix view of gene content overlap between these  modules, and you can see a few strongly overlapping group of modules, implicating similar pathology in different studies in different brain regions.

## Reading AMP-AD modules data

You can download data on the 30 human AMP-AD co-expression modules was obtained from the [Synapse data repository](https://www.synapse.org/Synapse:syn14237651). 

```{r AMP-AD1, warning=FALSE, message=FALSE}
query <- synTableQuery("SELECT * FROM syn11932957")
module_table <- read.table(query$filepath, sep = ",",header = TRUE)
```

Let's look at module table
```{r}
head(module_table)
```

Here you see 9 columns in this table. Columns we're interested in are:

*column 2: human ensembl gene ID, 

*column 3: module name in which gene is clustered and 

*column 7: is brain tissue.

*column 9: is gene names.

How many distinct modules are in table?
```{r}
length(unique(module_table$Module))
```

What are the name of modules?
```{r}
unique(module_table$Module)
```

How many genes are in each module?
```{r}
table(module_table$Module)
```

You can also visualize this as bar plot using ggplot2 package. 
```{r module_nGenes}
ggplot(module_table,aes(x=Module)) + 
  geom_bar() + 
  theme_bw() + 
  theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 0, size = 12))
```

> ## Challenge 1
> What are other ways to count genes in each module? 
>
> > ## Solution to Challenge 1
> >
> > ~~~
> > dplyr::count(module_table ,Module)
> > ~~~
> > {: .language-r}
> {: .solution}
{: .challenge}

We can also check total unique genes in table
```{r}
length(unique((module_table$GeneID)))
```


You can also check which modules come from which brain region.
```{r}
unique(module_table[c("Module", "brainRegion")])
```

#### Mouse-Human orthologous gene conversion
In the module table, we have human ENSEMBL ids and gene names. We will need corresponding mouse gene names to compare our results from mouse models. We are going to add mouse orthologous gene names corresponding to human ENSEMBL id. Mouse orthologs for human genes were extracted using the [HCOP](https://www.genenames.org/tools/hcop/) tool (The HGNC Comparison of Orthology Predictions) by [Wan, et al.](https://doi.org/10.1016/j.celrep.2020.107908). We are going to read that table from the [Synapse data repository](https://doi.org/10.7303/syn17010253.1)

```{r AMP-AD2, warning=FALSE, message=FALSE}
mouse.human.ortho <- fread(synapser::synGet("syn17010253")$path,check.names = F,header=T)
```

Let's see sometop rows of this ortholog table:
```{r}
head(mouse.human.ortho)
```

Add mouse gene names from ortholog table to module table by matching human ENSEMBL ids from both tables . 
```{r , warning=FALSE, message=FALSE}
module_table$Mouse_gene_name <- mouse.human.ortho$mouse_symbol[match(module_table$GeneID,mouse.human.ortho$human_ensembl_gene)]
```

```{r}
head(module_table)
```

We will only keep the columns of our interest and non-empty entries:
```{r}
ampad_modules <- module_table %>%
  distinct(tissue = brainRegion, module = Module, gene = GeneID, Mouse_gene_name) %>%
  filter(Mouse_gene_name != "")
```


```{r}
head(ampad_modules)
```

## Reading differential expression result of human data from meta-analysis
Differential expression meta-analysis of reprocessed RNASeq data from AMP-AD (all 7 brain regions). LogFC [values](https://www.synapse.org/#!Synapse:syn11180450) for human transcripts were obtained via the AMP-AD knowledge portal.  

```{r AMP-AD3, warning=FALSE, message=FALSE}
ampad_modules_raw <- fread(synapser::synGet("syn11180450")$path,check.names = F,header=T)
```

Let's check the data
```{r}
head(ampad_modules_raw)
```


Data came from how many tissues?
```{r}
unique(ampad_modules_raw$Tissue)
```
We see that data is from all 7 brain regions.

AMP-AD data has been processed many ways and using different models and comparisons. Let's see how many ways data has been analyzed,
```{r}
unique(ampad_modules_raw$Comparison)
```


For our analysis, we need data for the "Diagnosis" model and comparisons between case vs controls. We subset the logFC data for these conditions. We need only three columns: "Tissue","Gene" and "logFC".  So, we will filter and subset the data.
```{r AMP-AD4, warning=FALSE, message=FALSE}
ampad_fc <- ampad_modules_raw %>%
  as_tibble() %>%
  filter(Model == "Diagnosis", Comparison == "AD-CONTROL") %>%
  dplyr::select(tissue = Tissue, gene = ensembl_gene_id, ampad_fc = logFC)
```


#### Combine with modules so correlation can be done per module

Next, we will combine the fold change table `ampad_fc` and module table `ampad_modules`. First, look at both tables to check how can we merge them together?

```{r}
head(ampad_fc)
head(ampad_modules)
```
In both tables, common columns are "gene" and "tissue". So we will merge both datasets using these two columns. 
**Reminder**: Every gene can be present in multiple brain regions but only one module from per brain region. Let's check that:

```{r}
ampad_modules[ampad_modules$gene %in% "ENSG00000168439",]
```
We can see that this gene is present in six distinct modules and all modules are from different brain regions. You can do for any other gene as well.

Let's merge using `inner_join` function from tidyverse:
```{r AMP-AD5, warning=FALSE, message=FALSE}
ampad_modules_fc <- ampad_modules %>%
  inner_join(ampad_fc, by = c("gene", "tissue")) %>% 
  dplyr::select(symbol = Mouse_gene_name, module, ampad_fc) 
```


```{r}
head(ampad_modules_fc)
```
We will use `ampad_modules_fc` dataset to compare with log fold change data from mouse models.

#### Preparing module information for correlation plot 
```{r module}
mod <- c("TCXblue","PHGyellow","IFGyellow","DLPFCblue","CBEturquoise","STGblue","PHGturquoise","IFGturquoise","TCXturquoise","FPturquoise","IFGbrown","STGbrown","DLPFCyellow","TCXgreen","FPyellow","CBEyellow","PHGbrown","DLPFCbrown","STGyellow","PHGgreen","CBEbrown","TCXyellow","IFGblue","FPblue","FPbrown","CBEblue","DLPFCturquoise","TCXbrown","STGturquoise","PHGblue")

cluster_a <- tibble(
  module = c("TCXblue", "PHGyellow", "IFGyellow"),
  cluster = "Consensus Cluster A (ECM organization)",
  cluster_label = "Consensus Cluster A\n(ECM organization)"
)

cluster_b <- tibble(
  module = c("DLPFCblue", "CBEturquoise", "STGblue", "PHGturquoise", "IFGturquoise", "TCXturquoise", "FPturquoise"),
  cluster = "Consensus Cluster B (Immune system)",
  cluster_label = "Consensus Cluster B\n(Immune system)"
)

cluster_c <- tibble(
  module = c("IFGbrown", "STGbrown", "DLPFCyellow", "TCXgreen", "FPyellow", "CBEyellow", "PHGbrown"),
  cluster = "Consensus Cluster C (Neuronal system)",
  cluster_label = "Consensus Cluster C\n(Neuronal system)"
)

cluster_d <- tibble(
  module = c("DLPFCbrown", "STGyellow", "PHGgreen", "CBEbrown", "TCXyellow", "IFGblue", "FPblue"),
  cluster = "Consensus Cluster D (Cell Cycle, NMD)",
  cluster_label = "Consensus Cluster D\n(Cell Cycle, NMD)"
)

cluster_e <- tibble(
  module = c("FPbrown", "CBEblue", "DLPFCturquoise", "TCXbrown", "STGturquoise", "PHGblue"),
  cluster = "Consensus Cluster E (Organelle Biogensis, Cellular stress response)",
  cluster_label = "Consensus Cluster E\n(Organelle Biogenesis,\nCellular stress response)"
)

module_clusters <- cluster_a %>%
  bind_rows(cluster_b) %>%
  bind_rows(cluster_c) %>%
  bind_rows(cluster_d) %>%
  bind_rows(cluster_e) %>%
  mutate(cluster_label = fct_inorder(cluster_label))

head(module_clusters)

save(ampad_modules_fc,module_clusters,mod,file="../data/AMPADModuleData_Correlation.RData")

```


## Correlation between mouse models and human AD modules

There are two approaches that we adopted to compute correlation between mouse data with human AD modules:

* Compare change in expression in Human AD cases with change in expression in mouse models for each orthologous gene in a given module
    + LogFC(h) = log fold change in transcript expression of human AD patients compared to control patients. 
    + LogFC(m) = log fold change in transcript expression of mouse AD models compare to control mouse models.
        $$cor.test(LogFC(h), LogFC(m))$$


* Compare Human AD to mouse genetic effects for each orthologous gene in a given module
    + h = human gene expression (Log2 RNA-seq Fold Change control/AD)
    + Î² = mouse gene expression effect from linear regression model (Log2 RNA-seq TPM)
    $$cor.test(LogFC(h), Î²)$$


![Human-mouse logFCcorrelation](../fig/Picture4.png)


These appraoches allow us to assess directional coherence between AMP-AD modules and the effects of genetic perturbations in mice. In this lesson, we are going to use the first approach. 
Let's start!


#### Step 0: Reading Gene Expression Count matrix from Previous Lesson

We first read the result table saved after differential analysis in last [lesson](https://asliuyar.github.io/omicsTranslationAD/03-DESeq/index.html). We start with 5XFAD mouse model to understand all required steps to perform the correlation with human AD modules.
```{r FAD}
load("../results/DEAnalysis_5XFAD.Rdata")
```

We can also load AMP-AD module data.
```{r}
load("../data/AMPADModuleData_Correlation.RData")
```


#### Step 1: Measure the correlation between mouse models for each sex at each age and AMP-AD modules using common genes from both datasets

We compute a pearson correlation between changes in expression for each gene in a given module (log fold change for cases minus controls) with each mouse model (log fold change of the 5XFAD mice minus sex and age-matched B6 mice). 

First, we add both mouse `DE_5xFAD.df` and human `ampad_modules_fc` log fold change datasets for all genes.

```{r step1, message=FALSE, warning=FALSE}
model_vs_ampad <- DE_5xFAD.df %>%
    inner_join(ampad_modules_fc, by = c("symbol"),multiple = "all") 
```

```{r}
head(model_vs_ampad)
```

Next, we create a list-columns of data frame using [nest](https://tidyr.tidyverse.org/reference/nest.html) function of tidyverse package. Nesting is implicitly a summarising operation: you get one row for each group defined by the non-nested columns. 
```{r}
df <- model_vs_ampad %>%
        dplyr::select(module, model, sex, age, symbol, log2FoldChange, ampad_fc) %>%
        group_by(module, model, sex,age) %>%
        nest(data = c(symbol, log2FoldChange, ampad_fc))
```


```{r}
head(df)
```

Total number of groups in data table
```{r}
dim(df)
```

Let's check first row:
```{r}
head(df[1,]$data)
```

Next, we compute correlation coefficients using the cor.test function built in R as following:
```{r}
cor.df <- df  %>%
          mutate(
              cor_test = map(data, ~ cor.test(.x[["log2FoldChange"]], .x[["ampad_fc"]], method = "pearson")),
              estimate = map_dbl(cor_test, "estimate"),
              p_value = map_dbl(cor_test, "p.value")
              ) %>%
          ungroup() %>%
          dplyr::select(-cor_test)

head(cor.df)
```

#### Step 2: add signifcant correlations and join module cluster information to correlation table
```{r step2, message=FALSE, warning=FALSE}
model_module <- cor.df %>%
    mutate(significant = p_value < 0.05) %>%
    left_join(module_clusters, by = "module") %>%
    dplyr::select(cluster, cluster_label, module, model, sex,age,correlation = estimate, p_value, significant)
```

```{r}
head(model_module)
```

#### Step 3: Create a dataframe to use as input for plotting the results
```{r step3, message=FALSE, warning=FALSE}
order.model <- c("5xFAD (Male)", "5xFAD (Female)")

correlation_for_plot <- model_module %>%
    arrange(cluster) %>%
    mutate(
      module = factor(module,levels=mod),
      model_sex = glue::glue("{model} ({str_to_title(sex)})"),
    ) %>%
    arrange(model_sex) %>%
    mutate(
      model_sex = factor(model_sex,levels=order.model),
      model_sex = fct_rev(model_sex),
    )

head(correlation_for_plot)
```


### Visualizing the Correlation plot
Now, we will use the above matrix and visualize the correlation results using ggplot2 package.
```{r AMPAD1, message=FALSE, warning=FALSE, fig.width = 18, fig.height = 7, fig.show='hide'}

data <- correlation_for_plot
range(correlation_for_plot$correlation)

ggplot2::ggplot() +
    ggplot2::geom_tile(data = data, ggplot2::aes(x = .data$module, y = .data$model_sex), colour = "black", fill = "white") +
    ggplot2::geom_point(data = dplyr::filter(data), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = .data$correlation, size = abs(.data$correlation))) +
    ggplot2::geom_point(data = dplyr::filter(data, .data$significant), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = .data$correlation),color="black",shape=0,size=9) +
    ggplot2::scale_x_discrete(position = "top") +
    ggplot2::scale_size(guide = "none", limits = c(0, 0.4)) +
    ggplot2::scale_color_gradient2(limits = c(-0.4, 0.4), breaks = c(-0.4, 0, 0.4), low = "#85070C", high = "#164B6E", name = "Correlation", guide = ggplot2::guide_colorbar(ticks = FALSE)) +
    ggplot2::labs(x = NULL, y = NULL) +
    ggplot2::facet_grid( rows = dplyr::vars(.data$age),cols = dplyr::vars(.data$cluster_label), scales = "free", space = "free",switch = 'y') +
    ggplot2::theme(
      strip.text.x = ggplot2::element_text(size = 10,colour = c("black")),
      strip.text.y.left = ggplot2::element_text(angle = 0,size = 12),
      axis.ticks = ggplot2::element_blank(),
      axis.text.x = ggplot2::element_text(angle = 90, hjust = 0, size = 12),
      axis.text.y = ggplot2::element_text(angle = 0, size = 12),
      panel.background = ggplot2::element_blank(),
      plot.title = ggplot2::element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
      plot.title.position = "plot",
      panel.grid = ggplot2::element_blank(),
      legend.position = "right"
    )

```

```{r, echo=FALSE, warning=FALSE, include=FALSE}
png(file="../results/AMPAD1.png", width=15, height=6, units="in", res = 1200)
ggplot2::ggplot() +
    ggplot2::geom_tile(data = data, ggplot2::aes(x = .data$module, y = .data$model_sex), colour = "black", fill = "white") +
    ggplot2::geom_point(data = dplyr::filter(data), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = .data$correlation, size = abs(.data$correlation))) +
    ggplot2::geom_point(data = dplyr::filter(data, .data$significant), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = .data$correlation),color="black",shape=0,size=9) +
    ggplot2::scale_x_discrete(position = "top") +
    ggplot2::scale_size(guide = "none", limits = c(0, 0.4)) +
    ggplot2::scale_color_gradient2(limits = c(-0.4, 0.4), breaks = c(-0.4, 0, 0.4), low = "#85070C", high = "#164B6E", name = "Correlation", guide = ggplot2::guide_colorbar(ticks = FALSE)) +
    ggplot2::labs(x = NULL, y = NULL) +
    ggplot2::facet_grid( rows = dplyr::vars(.data$age),cols = dplyr::vars(.data$cluster_label), scales = "free", space = "free",switch = 'y') +
    ggplot2::theme(
      strip.text.x = ggplot2::element_text(size = 10,colour = c("black")),
      strip.text.y.left = ggplot2::element_text(angle = 0,size = 12),
      axis.ticks = ggplot2::element_blank(),
      axis.text.x = ggplot2::element_text(angle = 90, hjust = 0, size = 12),
      axis.text.y = ggplot2::element_text(angle = 0, size = 12),
      panel.background = ggplot2::element_blank(),
      plot.title = ggplot2::element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
      plot.title.position = "plot",
      panel.grid = ggplot2::element_blank(),
      legend.position = "right"
    )
dev.off()
```


![AMPAD - 5XFAD correlation plot](../results/AMPAD1.png)

In the above plot, the top row represents 30 AMP-AD modules grouped into 5 consensus clusters describing the major functional groups of AD-related alterations and the left column represents mouse models. Positive correlations are shown in blue and negative correlations in red. Color intensity and size of the circles are proportional to the correlation coefficient.  Black squares around dots represent significant correlation at p-value=0.05 and non-significant correlations are left blank. 

Male and female 5XFAD mice display gene expression alterations across all five consensus clusters, with the most pronounced alterations observed in Consensus Cluster B, which consists of immune system pathways. 

We also saw in the Volcano plot that immune-related genes were significantly overexpressed in the 5XFAD model. This significant positive correlation suggests that 5XFAD model captures inflammatory changes observed in human AD patients.

### Combing all steps into a function to perform correlation analysis

We can combine all the above steps performing correlation analyses into one function to use for other data by just inputting the log fold change matrix obtained from differential analysis.
```{r function, message=FALSE, warning=FALSE}
corr_function <- function(mydat)
{
  model_vs_ampad <- mydat %>%
    inner_join(ampad_modules_fc, by = c("symbol"),multiple = "all") %>%
    dplyr::select(module, model, sex, age, symbol, log2FoldChange, ampad_fc) %>%
    group_by(module, model, sex,age) %>%
    nest(data = c(symbol, log2FoldChange, ampad_fc)) %>%
    mutate(
      cor_test = map(data, ~ cor.test(.x[["log2FoldChange"]], .x[["ampad_fc"]], method = "pearson")),
      estimate = map_dbl(cor_test, "estimate"),
      p_value = map_dbl(cor_test, "p.value")
    ) %>%
    ungroup() %>%
    dplyr::select(-cor_test)
  
  model_module <- model_vs_ampad %>%
    mutate(significant = p_value < 0.05) %>%
    left_join(module_clusters, by = "module") %>%
    dplyr::select(cluster, cluster_label, module, model, sex,age,correlation = estimate, p_value, significant)

  correlation_for_plot <- model_module %>%
    arrange(cluster) %>%
    mutate(
      module = factor(module,levels=mod),
      model_sex = glue::glue("{model} ({str_to_title(sex)})"),
    ) %>%
    arrange(model_sex) %>%
    mutate(
      model_sex = factor(model_sex,levels=order.model),
      model_sex = fct_rev(model_sex),
    )
}
```


```{r function2, message=FALSE, warning=FALSE}
magora_corrplot <- function(data,ran) 
  {
  
  ggplot2::ggplot() +
    ggplot2::geom_tile(data = data, ggplot2::aes(x = .data$module, y = .data$model_sex), colour = "black", fill = "white") +
    ggplot2::geom_point(data = dplyr::filter(data), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = .data$correlation, size = abs(.data$correlation))) +
    ggplot2::geom_point(data = dplyr::filter(data, .data$significant), ggplot2::aes(x = .data$module, y = .data$model_sex, colour = .data$correlation),color="black",shape=0,size=9) +
    ggplot2::scale_x_discrete(position = "top") +
    ggplot2::scale_size(guide = "none", limits = c(0, ran)) +
    ggplot2::scale_color_gradient2(limits = c(-ran, ran), breaks = c(-ran, 0, ran), low = "#85070C", high = "#164B6E", name = "Correlation", guide = ggplot2::guide_colorbar(ticks = FALSE)) +
    ggplot2::labs(x = NULL, y = NULL) +
    #ggplot2::ggtitle("Model(Sex)| Age/Control") +
    ggplot2::facet_grid( rows = dplyr::vars(.data$age),cols = dplyr::vars(.data$cluster_label), scales = "free", space = "free",switch = 'y') +
    ggplot2::theme(
      strip.text.x = ggplot2::element_text(size = 10,colour = c("black")),
      strip.text.y.left = ggplot2::element_text(angle = 0,size = 12),
      axis.ticks = ggplot2::element_blank(),
      axis.text.x = ggplot2::element_text(angle = 90, hjust = 0, size = 12),
      axis.text.y = ggplot2::element_text(angle = 0, size = 12),
      panel.background = ggplot2::element_blank(),
      plot.title = ggplot2::element_text(angle = 0, vjust = -54, hjust = 0.03,size=12,face="bold"),
      plot.title.position = "plot",
      panel.grid = ggplot2::element_blank(),
      legend.position = "right"
    )
}
```

We can use these functions directly into our other dataset from the LOAD1 cohort. 

## Correlation between LOAD1 models and human AD modules

#### Reading Gene Expression Count matrix of LOAD1 cohort from Previous Lesson

```{r LOAD1}
load("../results/DEAnalysis_LOAD1.Rdata")
```

First, check the data: 
```{r}
head(DE_LOAD1.df)
```

```{r}
unique(DE_LOAD1.df$model)
unique(DE_LOAD1.df$sex)
unique(DE_LOAD1.df$age)
```

#### Perform the correlation analysis

We directly input the logFC table `DE_LOAD1.df` to `corr_function`:
```{r AMPAD2, message=FALSE, warning=FALSE}
order.model <- c("APOE4 (Male)", "APOE4 (Female)","Trem2 (Male)", "Trem2 (Female)","APOE4Trem2 (Male)", "APOE4Trem2 (Female)")

correlation_for_plot <- corr_function(DE_LOAD1.df)
```

```{r}
head(correlation_for_plot)
```

#### Visualize the correlation plot
```{r}
#range of correlation
range(correlation_for_plot$correlation)
```
```{r AMPAD_module_corPlot, fig.asp = 0.5, fig.width=16, fig.show='hide'}
magora_corrplot(correlation_for_plot,0.25)
```

```{r, echo=FALSE, warning=FALSE, include=FALSE}
png(file="../results/AMPAD_LOAD1.png", width=15, height=8, units="in", res = 1200)
magora_corrplot(correlation_for_plot,0.25)
dev.off()
```

![AMPAD - LOAD1 correlation plot](../results/AMPAD_LOAD1.png)

What do you conclude from this plot?
